edwards/ed25519/eddsa.go

```diff
@@ -0,0 +1,154 @@
+package ed25519
+
+import (
+	"crypto/cipher"
+	"crypto/sha512"
+	"errors"
+	"fmt"
+
+	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/nist"
+	"github.com/dedis/crypto/random"
+	"github.com/dedis/crypto/util"
+)
+
+type EdDSA struct {
+	Secret abstract.Secret
+	prefix []byte
```

Comment:
Not really. It's only used in EdDSA sign, because it's the algorithm definition.